polkadot/xcm/xcm-executor/src/tests/set_asset_claimer.rs

```diff
@@ -0,0 +1,139 @@
+    let xcm = Xcm::<TestCall>::builder_unsafe()
+        // if withdrawing fails we're not missing any corner case.
+        .withdraw_asset((Here, 100u128))
+        .clear_origin()
+        .pay_fees((Here, 10u128)) // 10% destined for fees, not more.
+        .build();
+
+    // We create an XCVM instance instead of calling `XcmExecutor::<_>::prepare_and_execute` so we
+    // can inspect its fields.
+    let mut vm =
+        XcmExecutor::<XcmConfig>::new(sender, xcm.using_encoded(sp_io::hashing::blake2_256));
+    vm.message_weight = XcmExecutor::<XcmConfig>::prepare(xcm.clone()).unwrap().weight_of();
+
+    let result = vm.bench_process(xcm);
+    assert!(result.is_ok());
+    assert_eq!(vm.asset_claimer(), None);
+}
+
+#[test]
+fn trap_then_set_asset_claimer() {
+    let sender = Location::new(0, [AccountId32 { id: [0; 32], network: None }]);
+    let bob = Location::new(0, [AccountId32 { id: [2; 32], network: None }]);
+
+    // Make sure the user has enough funds to withdraw.
+    add_asset(sender.clone(), (Here, 100u128));
+
+    // Build xcm.
+    let xcm = Xcm::<TestCall>::builder_unsafe()
+        // if withdrawing fails we're not missing any corner case.
+        .withdraw_asset((Here, 100u128))
+        .clear_origin()
+        .trap(0u64)
+        .set_asset_claimer(bob)
+        .pay_fees((Here, 10u128)) // 10% destined for fees, not more.
+        .build();
+
+    // We create an XCVM instance instead of calling `XcmExecutor::<_>::prepare_and_execute` so we
+    // can inspect its fields.
+    let mut vm =
+        XcmExecutor::<XcmConfig>::new(sender, xcm.using_encoded(sp_io::hashing::blake2_256));
+    vm.message_weight = XcmExecutor::<XcmConfig>::prepare(xcm.clone()).unwrap().weight_of();
+
+    let result = vm.bench_process(xcm);
+    assert!(result.is_err());
+    assert_eq!(vm.asset_claimer(), None);
+}
+
+#[test]
+fn set_asset_claimer_then_trap() {
+    let sender = Location::new(0, [AccountId32 { id: [0; 32], network: None }]);
+    let bob = Location::new(0, [AccountId32 { id: [2; 32], network: None }]);
+
+    // Make sure the user has enough funds to withdraw.
+    add_asset(sender.clone(), (Here, 100u128));
+
+    // Build xcm.
+    let xcm = Xcm::<TestCall>::builder_unsafe()
+        // if withdrawing fails we're not missing any corner case.
+        .withdraw_asset((Here, 100u128))
+        .clear_origin()
+        .set_asset_claimer(bob.clone())
+        .trap(0u64)
+        .pay_fees((Here, 10u128)) // 10% destined for fees, not more.
+        .build();
+
+    // We create an XCVM instance instead of calling `XcmExecutor::<_>::prepare_and_execute` so we
+    // can inspect its fields.
+    let mut vm =
+        XcmExecutor::<XcmConfig>::new(sender, xcm.using_encoded(sp_io::hashing::blake2_256));
+    vm.message_weight = XcmExecutor::<XcmConfig>::prepare(xcm.clone()).unwrap().weight_of();
+
+    let result = vm.bench_process(xcm);
+    assert!(result.is_err());
+    assert_eq!(vm.asset_claimer(), Some(bob));
+}
```

Comment:
```suggestion
}

```

An extra line at the end 😉